task-maker-async/src/store.rs

```diff
@@ -0,0 +1,114 @@
+#![allow(dead_code)]
+
+use serde::{Deserialize, Serialize};
+
+type HashData = Vec<u8>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InputFileHash(HashData);
+
+/// Two-level hash; the outer hash has information about all properties of a computation that are
+/// expected to change the result (such as outer hashes of inputs, or the command line). The inner
+/// hash takes care of properties of the computation that should not change the outputs, such as
+/// time and memory limits.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ComputationHash(HashData, HashData);
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InputFileWriteHandle(usize);
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ComputationWriteHandle(usize);
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ReadHandle(usize);
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ComputationDataFile {
```

Comment:
I don't understand what this enum is supposed to represent. Is OutputFile a full file (what is its name)? Is it the next chunk of output? What does Outcome mean? How does it interact with finalizing?

Maybe append_... is missing the argument with the actual data to append?

Reply:
Ah, yes, indeed there is a missing argument, and OutputFile(String) is just the filename. Outcome means the "file" representing the outcome (memory / time used, return code,  ...)